### PR TITLE
Update set-array syntax.

### PR DESCRIPTION
**Pull request by MikeDodaro**
Going Further ~ Generic Operations and Functions

### CODE REVIEW

File: articles/techniques/going-further.md [hunk 1]
```diff
@@ -46,15 +46,15 @@ A map from integers to Paulis, for instance, looks much the same as a map from f
 function MapIntsToPaulis(fn : Int -> Pauli, values : Int[]) : Pauli[] {
     mutable mappedValues = new Pauli[Length(values)];
     for (idx in 0..Length(values) - 1) {
-        set mappedValues[idx] = fn(values[idx]);
+        set mappedValues w/= idx <- fn(values[idx];
```
Comment (cgranade) [inline articles/techniques/going-further.md]: Looks like a closing parenthesis got dropped here?
```suggestion
        set mappedValues w/= idx <- fn(values[idx]);
```

File: articles/techniques/going-further.md [hunk 2]
```diff
@@ -46,15 +46,15 @@ A map from integers to Paulis, for instance, looks much the same as a map from f
...
     }
     return mappedValues;
 }
 
 function MapDoublesToStrings(fn : Double -> String, values : Double[]) : String[] {
     mutable mappedValues = new String[Length(values)];
     for (idx in 0..Length(values) - 1) {
-        set mappedValues[idx] = fn(values[idx]);
+        set mappedValues w/= idx <- fn(values[idx];
```
Comment (cgranade) [inline articles/techniques/going-further.md]:
```suggestion
        set mappedValues w/= idx <- fn(values[idx]);
```

File: articles/techniques/going-further.md [hunk 3]
```diff
@@ -83,7 +83,7 @@ For `Map`, we thus write:
 function Map<'Input, 'Output>(fn : 'Input -> 'Output, values : 'Input[]) : 'Output {
     mutable mappedValues = new 'Output[Length(values)];
     for (idx in 0..Length(values) - 1) {
-        set mappedValues[idx] = fn(values[idx]);
+        set mappedValues w/= idx <- fn(values[idx];
```
Comment (cgranade) [inline articles/techniques/going-further.md]:
```suggestion
        set mappedValues w/= idx <- fn(values[idx]);
```

File: articles/techniques/going-further.md (hunk 2 above)
Comment (bettinaheim) [reply articles/techniques/going-further.md]: oh god, I think I need to stop reviewing stuff on the side - thanks @cgranade!

**Pull request merged**